File: Data Structures & Algorithms/duplicate-integer/submission-0.java
// Time Complexity: O(n), single pass through array
// Space Complexity: O(n), hashset stores up to n unique elements

class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();

        for (int num : nums) {
            if (!hs.contains(num))  hs.add(num);
            else {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
# Time Complexity: O(n), single pass through list
# Space Complexity: O(n), hashmap stores up to n elements

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for i in range(len(nums)):
            diff: int = target - nums[i]
            if diff in hm:
                return [hm[diff], i]
            hm[nums[i]] = i
